test/NuGet.Clients.Tests/NuGet.PackageManagement.VisualStudio.Test/NetCorePackageReferenceProjectTests.cs

```diff
@@ -0,0 +1,522 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Xunit;
+using NuGet.VisualStudio;
+using NuGet.Test.Utility;
+using System.Threading;
+using System.IO;
+using Moq;
+using FluentAssertions;
+using NuGet.ProjectModel;
+using NuGet.Commands.Test;
+using NuGet.Commands;
+using NuGet.Configuration;
+using NuGet.Packaging.Core;
+using NuGet.Versioning;
+
+namespace NuGet.PackageManagement.VisualStudio.Test
+{
+    public class NetCorePackageReferenceProjectTests
```

Comment:
These tests look great. 

There's 1 scenario that we should test, which I was not able to find. 

1. For this scenario, imagine all packages are available and compatible.
1. PackageA has available versions, 1.0.0, 2.0.0
1. Prepare PackageA, version 1.0.0.
1. Simulate restore
1. Run GetInstalledPackages
1. Verify version is 1.0.0
1. Change the PackageReference version from 1.0.0 to 2.0.0
1. Simulate restore
1. Run GetInstlaledPackages
1. Verify version is 2.0.0

The idea behind this is to verify that changes in the assets file are detected.